### PR TITLE
Add SHA example

### PR DESCRIPTION
**Pull request by tzerrell**
Adds an example that computes the SHA-256 hash of `abc`.

### CODE REVIEW

File: sha/.gitignore [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.DS_Store
+Cargo.lock
+methods/guest/Cargo.lock
```
Comment (flaub) [inline sha/.gitignore:3]: I think lock files should probably be commited for apps (just not for libs): 
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

File: sha/host/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use methods::{HASH_ID, HASH_PATH};
+use risc0_zkp::core::sha::Digest;
+use risc0_zkvm::host::Prover;
+use risc0_zkvm::serde::{from_slice, to_vec};
+
+fn main() {
+    // Make the prover.
+    let method_code =
+        std::fs::read(HASH_PATH).expect("Method code should be present at the specified path");
```
Comment (flaub) [inline sha/host/src/main.rs:9]: Prefer `HASH_CONTENTS` so that you don't need disk I/O.

File: sha/host/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    let mut prover = Prover::new(&method_code, HASH_ID)
+        .expect("Prover should be constructed from matching code and method ID");
+
+    prover
+        .add_input(&to_vec("abc").expect("abc should serialize"))
+        .expect("Prover should accept input");
```
Comment (flaub) [inline sha/host/src/main.rs:15]: Would be neat to allow the string to be passed in as a command line argument.

**Review by flaub — APPROVED** [review]

**Comment by flaub** [comment]
Please add an entry to the `.github/workflows/main.yml` file to get some CI coverage.

**Comment by flaub** [comment]
Also would be good to have a test

File: sha/host/src/main.rs (hunk 2 above)
Comment (tzerrell) [reply sha/host/src/main.rs:10]: Hmm, is there a trick to this? Trying
```
use methods::{HASH_CONTENTS, HASH_ID, HASH_PATH};
```
gives the error
```
no `HASH_CONTENTS` in the root
```

Comment (flaub) [reply sha/host/src/main.rs:10]: Ah, this is probably only available on 0.12, sorry :)

**Pull request merged**